bcipy/tasks/rsvp/stopping_criteria.py

```diff
@@ -0,0 +1,231 @@
+    """ Returns true if the minimum number of iterations have not yet
+        been reached. """
+
+    def __init__(self, min_num_seq):
+        """ Args:
+                min_num_seq(int): minimum number of sequence number before any
+                 termination objective is allowed to be triggered """
+        self.min_num_seq = min_num_seq
+
+    def decide(self, epoch):
+        # Note: we use 'list_sti' parameter since this is the number of
+        # sequences displayed. The length of 'list_distribution' is 1 greater
+        # than this, since the language model distribution is added before
+        # the first sequence is displayed.
+        current_seq = len(epoch['list_sti'])
+        log.debug(
+            f"Checking min iterations; current iteration is {current_seq}")
+        return current_seq < self.min_num_seq
+
+
+class DecreasedProbabilityCriteria(DecisionCriteria):
+    """Returns true if the letter with the max probability decreased from the
+        last sequence."""
+
+    def decide(self, epoch):
+        if len(epoch['list_distribution']) < 2:
+            return False
+        prev_dist = epoch['list_distribution'][-2]
+        cur_dist = epoch['list_distribution'][-1]
+        return np.argmax(cur_dist) == np.argmax(
+            prev_dist) and np.max(cur_dist) < np.max(prev_dist)
+
+
+class MaxIterationsCriteria(DecisionCriteria):
+    """Returns true if the max iterations have been reached."""
+
+    def __init__(self, max_num_seq):
+        """ Args:
+                max_num_seq(int): maximum number of sequences allowed before
+                    mandatory termination """
+        self.max_num_seq = max_num_seq
+
+    def decide(self, epoch):
+        # Note: len(epoch['list_sti']) != len(epoch['list_distribution'])
+        # see MinIterationsCriteria comment
+        current_seq = len(epoch['list_sti'])
+        if current_seq >= self.max_num_seq:
+            log.debug(
+                "Committing to decision: max iterations have been reached.")
+            return True
+        return False
+
+
+class ProbThresholdCriteria(DecisionCriteria):
+    """Returns true if the commit threshold has been met."""
+
+    def __init__(self, threshold):
+        """ Args:
+                threshold(float \in [0,1]): A threshold on most likely
+                    candidate posterior. If a candidate exceeds a posterior
+                    the system terminates.
+                 """
+        assert 1 >= threshold >= 0, "stopping threshold should be in [0,1]"
+        self.tau = threshold
+
+    def decide(self, epoch):
+        current_distribution = epoch['list_distribution'][-1]
+        if np.max(current_distribution) > self.tau:
+            log.debug("Committing to decision: posterior exceeded threshold.")
+            return True
+        return False
+
+
+class MarginCriteria(DecisionCriteria):
+    """ Stopping criteria based on the difference of two most likely candidates.
+        This condition
```

Comment:
It looks like the documentation string is incomplete here?

Reply:
Documentation seems to be completed by Aziz.